File: platform/Leetcode/src/Containsduplicate.java
import java.util.HashMap;
import java.util.Scanner;

public class Containsduplicate {

    public static void main (String[] args) {
        System.out.println("Enter some integers separated by commas: ");
        Scanner input = new Scanner(System.in);
        String numbers = input.nextLine();
        boolean output = containsDuplicate(numbers);
        System.out.println("Do these numbers have duplicates? " + output);
    }
        
    public static boolean containsDuplicate(String numbers) {
        String[] digits = numbers.split(",");
        HashMap<String, Integer> collection = new HashMap<String,Integer>();

        for (String num:digits) {
            if (!collection.containsKey(num)) {
                collection.put(num, 1);
            } else {
                int valuePlusOne = collection.get(num) + 1;
                collection.put(num, valuePlusOne);
            }
        }
        boolean output = false; 
        for (int i=0; i < digits.length; i++) {
            if (collection.get(digits[i]) > 1) {
                output = true;
                break;
            }
        }
        return output;
    }
}

File: platform/Leetcode/src/contains_duplicate.py
def contains_duplicate(numbers, *args):
    assert isinstance(numbers, list), 'Input must be a list of integers'
    
    collect = {}
    for each in numbers:
        if each not in collect:
            collect[each] = 1
        else:
            collect[each] += 1
    for num in numbers:
        if collect[num] > 1:
            return True
    return False

def convert_to_list(string):
    return [int(each.strip()) for each in string.split(',') if each != '']

def main():
    user_input = input('Enter a list of numbers, separated by commas: ')
    numbers = convert_to_list(user_input)
    output = contains_duplicate(numbers)
    print("There are%sduplicates here." %(' ' if output else ' no '))
    
if __name__=='__main__':
    main()